routers/web/repo/pull.go

```diff
@@ -1501,7 +1501,7 @@ func CompareAndPullRequestPost(ctx *context.Context) {
 				return
 			}
 			ctx.Flash.Error(flashError)
-			ctx.JSONRedirect(pullIssue.Link()) // FIXME: it's unfriendly, and will make the content lost
+			ctx.JSONRedirect(ctx.Link + "?" + ctx.Req.URL.RawQuery) // FIXME: it's unfriendly, and will make the content lost
```

Comment:
Is it still unfriendly?

Reply:
Yes. See the description:

> This is only a quick fix to make it easier to backport.
> After this PR gets merged, I will propose a new PR to fix the FIXME.

Reply:
Yes, the filled form content will be lost because of the redirection. (The pull request title and content)
But it's the same behavior as before. So this can be resolved in future PR.

Reply:
->  Make PR form use toast to show error message #29545